src/fetchServices.ts

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env node
+
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {google} from 'googleapis';
+const spreadsheetId = '14JYzl_8W3HyD0c1jYDXQJA2sVWBZiMoxT8Sp57xGWvk';
+
+/**
+ * Make a call to the Google Cloud Service Management API's services.list endpoint
+ * @return {string[]} an array of hostnames ('foo.googleapis.com') for all available services
+ */
+export async function getAllServiceNames(): Promise<string[]> {
+  try {
+    const auth = await google.auth.getClient({
+      scopes: ['https://www.googleapis.com/auth/cloud-platform'],
+    });
+    const client = google.servicemanagement({
+      version: 'v1',
+      auth: auth,
+    });
+    const serviceNames: string[] = [];
+    return await listServices(client, serviceNames);
+  } catch (e) {
+    console.log(e);
+    throw e;
+  }
+}
+
+export async function listServices(
+  client: any,
```

Comment:
Guessing this is actually a type for this you want to use :) `any` is a crutch to be avoided.  Some notes on how to get at these types here:
https://github.com/googleapis/google-api-nodejs-client#typescript